alf/algorithms/mcts_algorithm.py

```diff
@@ -366,9 +433,40 @@ def predict_step(self, time_step: TimeStep, state: MCTSState):
                 path_lengths=path_lengths,
                 values=model_output.value)
 
-        action, info = self._select_action(trees, steps)
-        return AlgStep(
-            output=action, state=MCTSState(steps=state.steps + 1), info=info)
+    def _build_tree2(self, trees, to_plays):
+        """Build the tree by evaluating all the children of one node a time"""
+        branch_factor = trees.branch_factor
+        roots = (trees.B, trees.root_indices)
+        v = self._expand_children(trees, roots, 1, to_plays)
+        trees.visit_count[roots] += branch_factor
+        trees.value_sum[roots] += branch_factor * v
+        self._update_best_child(trees, roots)
```

Comment:
The logic of "expanding the roots' children and backing up" are identical to the `_build_tree1` case. I do not think the code duplication is a problem, but similar to above adding a comment line summarizing this code block can be helpful to the reader.

Reply:
added